advanced_source/static_quantization_tutorial.rst

```diff
@@ -0,0 +1,671 @@
+    torch.jit.save(torch.jit.script(float_model), saved_model_dir + scripted_float_model_file)
+
+  
+We see 78% accuracy on 300 images, a solid baseline for ImageNet, especially considering our model is just 14.0 MB. Measurnig the full accuracy takes about 5 minutes on a GPU.
+
+This will be our baseline to compare to. Next, let's try different quantization methods 
+
+4. Post-training static quantization  
+------------------------------------  
+
+Post-training static quantization involves not just converting the weights from float to int, 
+as in dynamic quantization, but also performing the additional step of first feeding batches  
+of data through the network and computing the resulting distributions of the different activations  
+(specifically, this is done by inserting `observer` modules at different points that record this  
+data). These distributions are then used to determine how the specifically the different activations  
+should be quantized at inference time (a simple technique would be to simply divide the entire range  
+of activations into 256 levels, but we support more sophisticated methods as well). Importantly,  
+this additional step allows us to pass quantized values between operations instead of converting these  
+values to floats - and then back to ints - between every operation, resulting in a significant speed-up.  
+
+.. code:: python
+
+    num_calibration_batches = 32
+
+    myModel = load_model(saved_model_dir + float_model_file).to('cpu')  
+    myModel.eval()  
+
+    # Fuse Conv, bn and relu  
+    myModel.fuse_model()  
+
+    # Specify quantization configuration  
+    # Start with simple min/max range estimation and per-tensor quantization of weights 
+    myModel.qconfig = torch.quantization.default_qconfig  
+    print(myModel.qconfig)  
+    torch.quantization.prepare(myModel, inplace=True) 
+
+    # Calibrate first 
+    print('Post Training Quantization Prepare: Inserting Observers')  
+    print('\n Inverted Residual Block:After observer insertion \n\n', myModel.features[1].conv) 
+
+    # Calibrate with the training set 
+    evaluate(myModel, criterion, data_loader, neval_batches=num_calibration_batches)  
+    print('Post Training Quantization: Calibration done') 
+
+    # Convert to quantized model  
+    torch.quantization.convert(myModel, inplace=True) 
+    print('Post Training Quantization: Convert done') 
+    print('\n Inverted Residual Block: After fusion and quantization, note fused modules: \n\n',myModel.features[1].conv) 
+
+    print("Size of model after quantization") 
+    print_size_of_model(myModel)  
+
+    top1, top5 = evaluate(myModel, criterion, data_loader_test, neval_batches=num_eval_batches) 
+    print('Evaluation accuracy on %d images, %2.2f'%(num_eval_batches * eval_batch_size, top1.avg))
+  
+For this quantized model, we see a significantly lower accuracy of just ~62% on these same 300  
```

Comment:
hi @raghuramank100 is this 67.3% on real dataset?

Reply:
Yes, we can remove the lower accuracy of ~62% and say " For this quantized model, we see an accuracy of 67.3% on the eval dataset